db/versions/0017-migration.sql

```diff
@@ -0,0 +1,42 @@
+begin
+
+  -- lock this table before reading from it, to prevent loss of concurrent
+  -- updates when the table is dropped.  Note that this may lead to concurrent
+  -- updates failing; the important thing is that they not succeed without
+  -- taking effect.  Failed updates will be retried.
+  lock table taskcluster_github_builds_entities;
+
+  create table github_builds
+  as
+    select
+      (value ->> 'organization')::text as organization,
+      (value ->> 'repository')::text as repository,
+      (value ->> 'sha')::text as sha,
+      (value ->> 'taskGroupId')::text as task_group_id,
+      (value ->> 'state')::text as state,
+      (value ->> 'created')::timestamptz as created,
+      (value ->> 'updated')::timestamptz as updated,
+      (value ->> 'installationId')::integer as installation_id,
+      (value ->> 'eventType')::text as event_type,
+      (value ->> 'eventId')::text as event_id,
+      etag
+    from taskcluster_github_builds_entities;
+  alter table github_builds add primary key (task_group_id);
+  alter table github_builds
+    alter column organization set not null,
+    alter column repository set not null,
+    alter column sha set not null,
+    alter column task_group_id set not null,
+    alter column state set not null,
+    alter column created set not null,
+    alter column updated set not null,
+    alter column installation_id set not null,
+    alter column event_type set not null,
+    alter column event_id set not null,
+    alter column etag set not null,
+    alter column etag set default public.gen_random_uuid();
+
```

Comment:
Since `get_github_builds` and `github.builds(..)` can filter by any of organization, repository, and sha, it probably makes sense to add indexes for each of those columns -- otherwise we'll be doing a full table scan for any request there.  The table is only 37240 rows at the moment, but doesn't expire, so that could get pretty expensive.

Reply:
Did this get missed?

Reply:
Ah yeah

Reply:
I added an index on `(org, repo, sha)` since although you're allowed to just search for a sha or a repo without an org, you generally won't want to.

Reply:
Makes sense.  We should disallow that, then -- either in the db fn or in the API method.  We could do that in a followup if you'd prefer.